testdata/event_listener

```diff
@@ -169,7 +169,7 @@ compact         1   2.3 K             0 B          (size == estimated-debt, in =
 zmemtbl         0     0 B
    ztbl         0     0 B
  bcache         8   1.4 K    5.9%  (score == hit-rate)
- tcache         1   616 B    0.0%  (score == hit-rate)
+ tcache         1   640 B    0.0%  (score == hit-rate)
```

Comment:
if the fact that `Reader` got bigger by a slice header is bad, we could potentially just have `maybeSwapSuffix` just go read the UserProps map every time instead of doing it once on creation but my guess was we'd rather not do the whole map lookup to read every block